pkg/deploy/api/types.go

```diff
@@ -182,6 +182,14 @@ type DeploymentTriggerImageChangeParams struct {
 	From kapi.ObjectReference `json:"from"`
 	// Tag is the name of an image repository tag to watch for changes.
 	Tag string `json:"tag,omitempty"`
+	// Status contains information about the last trigger event.
+	Status DeploymentTriggerImageChangeParamsStatus `json:"status"`
```

Comment:
Flatten this and add a TODO in BuildConfig to rename to LastTriggeredImage.  And it's not really an ID, it's the image's  name or unique identifier (but not uid).